src/test/kotlin/com/terraformation/backend/tracking/db/ObservationResultsStoreTest.kt

```diff
@@ -264,14 +264,37 @@ class ObservationResultsStoreTest : DatabaseTest(), RunsAsUser {
         sizeMeters: Int,
     ) {
       importSiteFromCsvFile(prefix, sizeMeters)
-      val summaryBeforeObservations = resultsStore.fetchSummaryForPlantingSite(plantingSiteId)
-      val summaries =
+      val summaryBeforeObservations = resultsStore.fetchSummariesForPlantingSite(plantingSiteId)
+
+      val observationTimes =
           List(numObservations) {
-            importObservationsCsv(prefix, numSpecies, it)
-            resultsStore.fetchSummaryForPlantingSite(plantingSiteId)!!
+            val time = Instant.ofEpochSecond(it.toLong())
+            importObservationsCsv(prefix, numSpecies, it, time) to time
           }
-      assertNull(summaryBeforeObservations, "No observations made yet.")
-      assertSummary(prefix, summaries)
+
+      val summaries = resultsStore.fetchSummariesForPlantingSite(plantingSiteId)
+      assertEquals(
+          emptyList<ObservationRollupResultsModel>(),
+          summaryBeforeObservations,
+          "No observations made yet.")
+      assertSummary(prefix, summaries.filterNotNull().reversed())
+
+      assertEquals(
+          summaries.take(2),
+          resultsStore.fetchSummariesForPlantingSite(plantingSiteId, limit = 2),
+          "Partial summaries via limit should contain the latest observations.")
```

Comment:
The beginning of the list should have the most recent summary. Since observations were fetched ordered descending by completion time. Assuming maxCompletionTime is not set, the first entry will use completed observations up to the latest,  second will use up to second latest, etc. until will hit the limit